### PR TITLE
Fixes Avro der/ser with array objects

### PR DESCRIPTION
**Pull request by rbalaban-mdsol**
This branch is merged with `feature/add-kafka-info-headers` it addresses the issues with Avro der/ser. 

For deserializing, it checks for `java.util.List` and loops over the items calling `toSpecific`, this removes errors when deserializing array messages in Avro. 

Serializing to Kafka works as expected, Jackson was throwing errors since, we call `.toJava` on the arrays ( to be compatible with Avro ), this wraps the objects. This applies the same functionality from `Arrays` after converting `.toScala`.

### CODE REVIEW

File: src/main/scala/io/epiphanous/flinkrunner/serde/Codec.scala [hunk 1]
```diff
@@ -52,10 +52,6 @@ case class Codec[E](
         jsonConfig.sortKeys
       )
       .configure(SerializationFeature.INDENT_OUTPUT, jsonConfig.pretty)
-      .configure(
```
Comment (nextdude) [inline src/main/scala/io/epiphanous/flinkrunner/serde/Codec.scala:55]: Why did you remove this?

File: src/main/scala/io/epiphanous/flinkrunner/util/AvroUtils.scala [hunk 2]
```diff
@@ -121,6 +121,29 @@ object AvroUtils extends LazyLogging {
         .foldLeft(instance) { (a, field) =>
           val f = field.name()
           genericRecord.get(f) match {
+            case array: java.util.List[_] =>
```
Comment (nextdude) [inline src/main/scala/io/epiphanous/flinkrunner/util/AvroUtils.scala:124]: Feels like we should DRY up this block with the similar code starting at L148.

**Review by nextdude — CHANGES_REQUESTED** [review]
Looks OK, except for one question and one small DRY refactor.

File: src/main/scala/io/epiphanous/flinkrunner/util/AvroUtils.scala (hunk 2 above)
Comment (rbalaban-mdsol) [reply src/main/scala/io/epiphanous/flinkrunner/util/AvroUtils.scala:144]: Agreed, DRYed up the implementation.

**Review by nextdude — APPROVED** [review]
LGTM
